var/spack/repos/builtin/packages/amrex/package.py

```diff
@@ -241,3 +244,55 @@ def cmake_args(self):
             args.append('-DCUDA_ARCH=' + self.get_cuda_arch_string(cuda_arch))
 
         return args
+
+    # TODO: Replace this method and its 'get' use for cmake path with
+    #   join_path(self.spec['cmake'].prefix.bin, 'cmake') once stand-alone
+    #   tests can access build dependencies through self.spec['cmake'].
+    def cmake_bin(self, set=True):
+        """(Hack) Set/get cmake dependency path."""
+        filepath = join_path(self.install_test_root, 'cmake_bin_path.txt')
+        if set:
+            with open(filepath, 'w') as out_file:
+                cmake_bin = join_path(self.spec['cmake'].prefix.bin, 'cmake')
+                out_file.write('{0}\n'.format(cmake_bin))
+        else:
+            with open(filepath, 'r') as in_file:
+                return in_file.read().strip()
+
+    @run_after('build')
+    def setup_smoke_test(self):
+        """Skip setup smoke tests for AMReX versions less than 21.12."""
+        if self.spec.satisfies('@:21.11'):
+            return
+
+        self.cache_extra_test_sources(['Tests'])
+
+        # TODO: Remove once self.spec['cmake'] is available here
+        self.cmake_bin(set=True)
+
+    def test(self):
+        """Skip smoke tests for AMReX versions less than 21.12."""
+        if self.spec.satisfies('@:21.11'):
+            print("SKIPPED: Stand-alone tests not supported for this version of AMReX.")
+            return
+
+        """Perform smoke tests on installed package."""
+        # TODO: Remove/replace once self.spec['cmake'] is available here
+        cmake_bin = self.cmake_bin(set=False)
+
+        args = []
+        args.append('-S ./cache/amrex/Tests/SpackSmokeTest')
```

Comment:
The build fails for me without removing the space between `-S` and the path as follows:

```suggestion
        args.append('-S./cache/amrex/Tests/SpackSmokeTest')
```

Reply:
That is curious, which CMake version and system? I have never seen this fail :)

Reply:
Version `3.14.5`.  It's a system-installed package that 1) is the default for me on that system and 2) matches the version used during package install.

Reply:
I see. I use 3.18+ in all my software and I seem to remember they reworked their CLI argument parser at some point.
We should probably bump AMReX in the near future, too https://github.com/AMReX-Codes/amrex/blob/22.01/CMakeLists.txt#L1 (and get rid of CUDA & HIP work-arounds for these old CMake releases.)

Anyhow, good for now :)

Reply:
I made this change as suggested.